include/seqan3/range/views/pseudo_random_access.hpp

```diff
@@ -0,0 +1,199 @@
+ */
+/*!\fn pseudo_random_access_range::end_ra()
+ * \brief Returns a pseudo random access sentinel.
+ *
+ * \details
+ *
+ * The returned sentinel must model std::sentinel_for with the corresponding iterator obtained by begin_ra().
+ *
+ * \attention This is a concept requirement, not an actual function (however types
+ *            modelling this concept will provide an implementation).
+ */
+//!\cond
+template <typename rng_t>
+SEQAN3_CONCEPT pseudo_random_access_range = std::ranges::range<rng_t> && requires (rng_t rng)
+{
+    rng.begin_ra();
+    rng.end_ra();
+
+    requires std::random_access_iterator<decltype(rng.begin_ra())>;
+    requires std::sentinel_for<decltype(rng.begin_ra()), decltype(rng.end_ra())>;
+};
+//!\endcond
+
+// ============================================================================
+//  pseudo_random_access_fn (adaptor definition)
+// ============================================================================
+
+//!\brief View adaptor definition for seqan3::views::pseudo_random_access.
+struct pseudo_random_access_fn : public adaptor_base<pseudo_random_access_fn>
+{
+private:
+    //!\brief Type of the CRTP-base.
+    using base_t = adaptor_base<pseudo_random_access_fn>;
+
+public:
+    //!\brief Inherit the base class's Constructors.
+    using base_t::base_t;
+
+private:
+    //!\brief Befriend the base class so it can call impl().
+    friend base_t;
+
+    /*!\brief Call the view's constructor with the underlying view as argument.
+     * \returns An instance of the adapted range.
+     */
+    template <std::ranges::viewable_range urng_t>
+    static constexpr auto impl(urng_t && urange)
+    {
+        static_assert(std::ranges::random_access_range<urng_t> || pseudo_random_access_range<urng_t>,
+                      "The adapted range must either model std::ranges::random_access_range or must be "
+                      "a specific SeqAn range type that supports pseudo random access.");
+
+        if constexpr (std::ranges::random_access_range<urng_t>)
+        { // Nothing to do, just return as ref_view or original view.
+            return std::views::all(std::forward<urng_t>(urange));
+        }
+        else
+        { // Get a subrange using the random access iterators of the container.
+            using iterator_ra_t = decltype(std::declval<urng_t>().begin_ra());
+            using sentinel_ra_t = decltype(std::declval<urng_t>().end_ra());
+
+            return std::ranges::subrange<iterator_ra_t, sentinel_ra_t>{urange.begin_ra(), urange.end_ra()};
+        }
+    }
+};
+
+}  // namespace seqan3::detail
+
+namespace seqan3::views
+{
+
+/*!\name General purpose views
+ * \{
+ */
+
+/*!\brief            A view adaptor that converts a pseudo random access range to a std::random_access_range.
+ * \tparam urng_t    The type of the range being processed. See below for requirements. [template parameter is
+ *                   omitted in pipe notation]
```

Comment:
```suggestion
 *                   omitted in pipe notation].
```

Reply:
Sure, that is like copy'n'paste and then needs to be fixed everywhere.

Reply:
maybe Add this to the documentation to do issue :)

Reply:
Well, I guess that was intentional without the dot. See it is everywhere in the view documentation. The dot is before and this is kind of an additional note for the reader.

Reply:
ok